iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/ResourceDeclarationConverter.java

```diff
@@ -69,6 +75,26 @@ public ResourceDeclaration convertToResourceDeclaration(MappingTree tree) {
       .map(this::toProperties)
       .orElse(Collections.emptyList());
 
-    return new ResourceDeclarationImpl(name, version, type, otherProperties);
+    Optional<PropertyTree> optChildResources = PropertyUtils.get(tree, "resources");
+    if (optChildResources.isPresent()) {
+      List<ResourceDeclaration> childResources = optChildResources
+        .map(PropertyTree::value)
+        .filter(SequenceTree.class::isInstance)
+        .map(SequenceTree.class::cast)
+        .map(sequenceTree -> mappingTreeOnly(sequenceTree.elements()))
+        .map(m -> m.stream().map(mappingTree -> convertToResourceDeclaration(mappingTree, buildParentType(parentType, type))).collect(Collectors.toList()))
+        .orElse(Collections.emptyList());
+      return new GroupResourceDeclarationImpl(name, version, type, parentType, otherProperties, childResources);
+    } else {
+      return new ResourceDeclarationImpl(name, version, type, parentType, otherProperties);
+    }
```

Comment:
My suggestion would be to use ifPresentOrElse and call another method to build the GroupResourceDeclaration.

Reply:
I failed to come up with a solution with `ifPresentOrElse`, I don't see how it should be. Could you provide more insight? :sweat_smile:

Reply:
I came up with an approach map(toResourceGroupDeclaration(...)).orElseGet(toResourceDeclaration(...)) approach with separate methods to build the object. Let me know if you had something else in mind.